### PR TITLE
Add icon title

### PR DESCRIPTION
**Pull request by czebe**
- Added support for `title` prop (text or React node)
- Added propTypes
- Removed `display: inline-block`, this should probably go into the `tw-icon` class

### CODE REVIEW

File: lib/react/buildModules.js [hunk 1]
```diff
@@ -53,18 +53,38 @@ function buildModuleForNameWithSvgAndClass(name, svg, className) {
...
 
-export default (props) => (
+const Icon = ({ size, className, title, style }) => (
   <span
-    {...props}
-    className={\`${className}\${ props.size ? ' ${className}-' + props.size : ''} \${ props.className || '' }\`}
-    style={{ ...props.style, display: 'inline-block' }}
+    className={\`${iconClassName}\${ size ? ' ${iconClassName}-' + size : ''} \${ className ? ' ' + className : '' }\`}
+    style={{ ...style }}
+    aria-hidden={!title ? 'true' : null}
```
Comment (oliverviljamaa) [inline lib/react/buildModules.js]: Does this result in `aria=hidden="null"` or no `aria-hidden` attribute when there's no title?
If the former, then why not just have `aria-hidden={!!title}`?

File: lib/react/buildModules.js [hunk 2]
```diff
@@ -53,18 +53,38 @@ function buildModuleForNameWithSvgAndClass(name, svg, className) {
...
+import PropTypes from 'prop-types';
 
-export default (props) => (
+const Icon = ({ size, className, title, style }) => (
   <span
-    {...props}
-    className={\`${className}\${ props.size ? ' ${className}-' + props.size : ''} \${ props.className || '' }\`}
-    style={{ ...props.style, display: 'inline-block' }}
+    className={\`${iconClassName}\${ size ? ' ${iconClassName}-' + size : ''} \${ className ? ' ' + className : '' }\`}
+    style={{ ...style }}
```
Comment (oliverviljamaa) [inline lib/react/buildModules.js]: Don't we lose the `display: 'inline-block';` that we had here?

File: lib/react/buildModules.js (hunk 1 above)
Comment (czebe) [reply lib/react/buildModules.js]: It results in no `aria-hidden` attribute defined on the tag when there's a title supplied.
So either: `<span aria-hidden="true" ...>` or `<span ...>`

File: lib/react/buildModules.js (hunk 2 above)
Comment (czebe) [reply lib/react/buildModules.js]: `display: 'inline-block';` should go into `.tw-icon` base class. If we keep it here, there's no easy way to override it (eg. have a `display: block;` icon).

We could invert the order to allow overriding (`{{ display: 'inline-block', ...props.style }}`), but I feel it's better to extract this to the base icon class.

Comment (oliverviljamaa) [reply lib/react/buildModules.js]: Cool, can't remember why I didn't do that... Maybe to have less to edit manually for the consumer, in our case `@transferwise/icons`.

Comment (oliverviljamaa) [reply lib/react/buildModules.js]: I think it was actually that, I think I'd prefer allowing overriding, but having a sensible default, even without the custom styles. WDYT?

Comment (czebe) [reply lib/react/buildModules.js]: Does `inline-block` make a difference? I'm not even sure...
`<span>` will be inline, is that not good enough for defaults?

Comment (oliverviljamaa) [reply lib/react/buildModules.js]: I guess we need it for the `width` and `height`, and we need those only for the custom classes, so I guess it's fine to have the `display` style in `icons` as well.

**Pull request merged**